src/main/java/com/example/bookshop/controller/BookController.java

```diff
@@ -45,8 +47,10 @@ public void delete(@PathVariable Long id) {
         bookService.deleteById(id);
     }
 
+    @ResponseStatus(HttpStatus.ACCEPTED)
     @PutMapping("/{id}")
-    public BookDto update(@RequestBody CreateBookRequestDto requestDto, @PathVariable Long id) {
+    public BookDto update(@RequestBody @Valid CreateBookRequestDto requestDto,
+                          @PathVariable Long id) {

```

Comment:
you may also validate id here:
```
@PathVariable @Positive Long id
```

in order to do so - annotate your controller with `@Validated`